### ISSUE TITLE
Enable pricing for GCP custom instances

### ISSUE THREAD

**Issue by 7onn** [infracost/infracost#1447]
**What do you like to be added and why**?
I'd like to enable Infracost to estimate pricing for custom instances on GCP (e.g: custom-20-20480 with 20 CPUs and 20GB of ram).

A few of the GKE node pools of ours are relying on customized instances like the above-mentioned and this is generating warnings on the breakdown instead of estimating their actual cost.

```
time="2022-03-09T16:02:49Z" level=warning msg="No products found for google_container_node_pool.REDACTED Instance usage (Linux/UNIX, on-demand, custom-32-131072), using 0.00"
time="2022-03-09T16:02:49Z" level=warning msg="No products found for google_container_node_pool.REDACTED Instance usage (Linux/UNIX, preemptible, custom-22-20480), using 0.00"
time="2022-03-09T16:02:49Z" level=warning msg="No products found for google_container_node_pool.REDACTED Instance usage (Linux/UNIX, preemptible, custom-22-20480), using 0.00"
time="2022-03-09T16:02:49Z" level=warning msg="No products found for google_container_node_pool.REDACTED Instance usage (Linux/UNIX, on-demand, custom-32-131072), using 0.00"
```

I first detected this issue in a resource like this:
```terraform
resource "google_container_node_pool" "dev-1" {
  cluster  = "dev"
  name     = "dev-1"
  location = "europe-central2-c"

  autoscaling {
    max_node_count = 20
    min_node_count = 1
  }

  management {
    auto_repair  = true
    auto_upgrade = true
  }

  upgrade_settings {
    max_surge       = 1
    max_unavailable = 0
  }

  node_config {
    preemptible = true
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform"
    ]

    machine_type = "custom-20-20480"

    image_type   = "COS_CONTAINERD"
    disk_type    = "pd-standard"
    disk_size_gb = "100"

    labels = {
      "pool" = "dev-1"
    }

    workload_metadata_config {
      mode = "GKE_METADATA"
    }
  }
}

``` 

**Notes**:
I'd enjoy developing such a feature myself. So, any guidance for where I should start looking would be most welcome. Anyway, I'm planning to take this weekend to dive deeper into Infracost's codebase.

**Comment by alikhajeh1** [comment]
@7onn are you still interested in working on this? if so, feel free to join https://www.infracost.io/community-chat so we can guide you

**Comment by 7onn** [comment]
Sure! I'll join the slack channel to seek orientation about this =)

I think this would involve not only the Infracost CLI code but also the Cloud pricing API, right? =)

**Comment by aliscott** [comment]
@antonym, @slarimore02, @Praveen156 support for custom GCP instances has now been released in v0.10.14.

Thank you @7onn for contributing this! 🚀 🙏 🎉

**Comment by 7onn** [comment]
Thanks for the support for doing it so!
I thrilled with this latest and impactful contribution \o/
